test/gtest/uct/uct_test.cc

```diff
@@ -209,40 +214,48 @@ void uct_test::set_interface_rscs(uct_component_h cmpt, const char *name,
 {
     int i;
 
-    /* Create two resources on the same interface. the first one will have the
-     * ip of the interface and the second one will have INADDR_ANY */
-    for (i = 0; i < 2; i++) {
+    /* Create three resources on the same interface:
+     *  0 - has the ip of the dst interface
+     *  1 - has the ip of the dst interface and IP of src interface
+     *  2 - has INADDR_ANY
+     */
+    for (i = 0; i < 3; i++) {
         resource rsc(cmpt, std::string(name), local_cpus, "sockaddr",
                      std::string(ifa->ifa_name), UCT_DEVICE_TYPE_NET);
+        bool init_src_addr = (i == 1);
```

Comment:
move this to 228

Reply:
depends on https://github.com/openucx/ucx/pull/6007/files#r543409299

Reply:
IMO better localize it and move to the relevant if part only

Reply:
now it's closer to comment before the loop, so it's simpler to maintain them in consistency